Pep8 insanity

bdowling opened the pull request:
Let's finally bring things up to pep8 after #62 is merged (didn't want to pollute diff in that PR).

lockefox commented on Robinhood/Robinhood.py:
```diff
@@ -1,37 +1,18 @@
 from six.moves.urllib.request import getproxies
 from six.moves import input
 
+from enum import Enum
```
Enum is not external in py3.

Py2 is given as a courtesy, but we should assume 3.5+ for style

lockefox commented on Robinhood/Robinhood.py:
```diff
@@ -73,11 +57,77 @@ class Robinhood:
 
+    from . import exceptions
+
+    class Bounds(Enum):
```
Enums moved inside class?  Why?  Won't this break tests since enums are no longer referenceable directly?

lockefox commented on Robinhood/Robinhood.py:
```diff
@@ -128,24 +176,21 @@ def login(self,
-        except requests.exceptions.HTTPError:
-            raise RH_exception.LoginFailed()
+        res = self.session.post(self.endpoints['login'], data=payload)
+        self.raiseForStatus(res)
```
This couple's hard with another PR.  Merge should be able to handle this.

Also if that PR changes significantly, this PR will need adjustment.

Last, we remove the chance to raise a library-specific exception here that should be useful

lockefox commented:
I have a lot of concerns in the updates here.  There are many bigger changes than just 'PEP".  These could cause a lot of pain for users.  Also I am not sure the test suite was run over this b/c there should be some big namespace bugs causing test failures

lockefox commented on setup.py:
```diff
@@ -75,9 +76,10 @@ class PyTest(TestCommand):
             '--cov=Robinhood/',
-            '--cov-report=term-missing'
+            '--cov-report=term-missing',
+	    '--pep8'
```
This breaks?  Does pyflakes or a plugin need to be added to the `test_requires`?

bdowling commented:
@lockefox To be honest.. I was not running the tests consistently before, there were a number that were failing and it looks to me like they have also been failing on master.  I didn't catch the pep8 vs pytest_pep8 requirement because of my test env had the requirements.  I fixed that.  I know flake8 may be more preferred, we could certainly switch to that.

However I spent some time tonight and fixed a lot of these errors.  Some of them required changes to the test as the API was returning results in a different way than expected, but I checked my code against master and believe they were working the same, so the tests needed fixing.  I welcome a set of eyes on these changes just to make sure I didn't fix tests for impacts of this branch.

I did drop my tests into a copy of master to make sure they faired ok.

One of my tests did change to passing 'symbol' into instruments to prevent the query search that was being performed (which was not exact enough to find the MU symbol, causing the prior failure in that test case..)

The only other exception to this is the custom exceptions (no pun intended).  In a few cases my new exception handing did change the custom exceptions to HTTPError.  I think this is an open topic we should discuss how best to handle, and is likely a case-by-case issue, but could be addressed after merging this PR IMHO. 

Yes, this branch as mentioned was layered on top of #62.  I just didn't want to conflate the diff in that PR with all the pep8 stuff, but at same time I know pep8 is important to many folks.   So either this PR is preferred or these new fixes would need to be merged back.

I understand breaking the API for people is not preferred, thus why I specifically mentioned the proposed does in a few minor cases.  However I'm not sure at this point how many people have "large implementations" that could not adapt to us improving things.  With the issues that have been addressed, this library is approaching more stability, so trying to change things to "cleanly stabilize" the overall API/lib is not really a bad thing.

Cheers!

bdowling commented:
Closing this PR, changes have been merged into #62.  

Note the message above about all of the tests now pass..  I saw the #64 was created about broken tests.  Those should be fixed now.
